x-pack/plugins/ml/server/saved_objects/repair.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from '@hapi/boom';
+import { IScopedClusterClient } from 'kibana/server';
+import type { JobObject, JobSavedObjectService } from './service';
+import { JobType } from '../../common/types/saved_objects';
+import { checksFactory } from './checks';
+
+import { Datafeed } from '../../common/types/anomaly_detection_jobs';
+
+export function repairFactory(
+  client: IScopedClusterClient,
+  jobSavedObjectService: JobSavedObjectService
+) {
+  const { checkStatus } = checksFactory(client, jobSavedObjectService);
+
+  async function repairJobs(simulate: boolean = false) {
```

Comment:
what about having the`simulate` check on top and return the default object instead of checking in teach for loop?